mantidproject/mantid
# Add PDFgetN style normalization to SumSpectra

This adds an option (default keeps old behavior) to multiply the output workspace by the number of spectra contributing. The reason for this option is that in PDFgetN, the data is normalized and multiplying by the number of contributing spectra does not preserve the correct behavior of the data.

*Report to:* @marshallmcdonnell

**To test:**

Run the following script and see the difference (most visible in low-Q) between `FkQ` and `FkQ_old`. Setting `MultiplyBySpectra=False` gives the appropriate summing for normalized data.
```python
import numpy as np

filename = 'Si_SRM_e_d91772.nxs'
# 27 appears to be S(Q) - 1
LoadNexusProcessed(Filename=filename, OutputWorkspace='banked', EntryNumber=27)

# reduce range
CropWorkspaceRagged(InputWorkspace='banked', OutputWorkspace='cropped',
                    Xmin=[0.67, 1.20, 2.42, 3.70, 4.12, 0.39],
                    Xmax=[10.20, 20.8, np.nan, np.nan, np.nan, 9.35])

# bin everything on a constant scale for summing
binning=(0.,.02,40.)
Rebin(InputWorkspace='cropped', OutputWorkspace='cropped',
          Params=binning)

# put into a single spectrum and Fourier transform
SumSpectra(InputWorkspace='cropped', OutputWorkspace='FkQ_old',
           WeightedSum=True, RemoveSpecialValues=True)
SumSpectra(InputWorkspace='cropped', OutputWorkspace='FkQ',
           WeightedSum=True, RemoveSpecialValues=True, MultiplyBySpectra=False)

PDFFourierTransform(InputWorkspace='FkQ', OutputWorkspace='Gr',
                    QMin=.67, # from above
                    InputSofQType='S(Q)-1', DeltaR=.02)
```

Fixes #21773.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
